frame/support/src/storage/unhashed.rs

```diff
@@ -154,6 +154,10 @@ pub fn clear_prefix(
 	MultiRemovalResults { maybe_cursor, backend: i, unique: i, loops: i }
 }
 
+pub fn contains_prefix(prefix: &[u8]) -> bool {
+	sp_io::storage::next_key(prefix).is_some()
```

Comment:
Is this the way to go @cheme ?

Reply:
need also to check that the prefix is part of the next_key.
think it may fail when K2 is [] also (will think there is no key in prefix).
To implement with next_key is not very straight forward here, probably some thing like.
if k1.is_empty {
 if io::contains([], []) {
 return true
}
}
let prev_prefix = k1.clone()
prev_prefix.pop();
prev_prefix.push(255);
if let Some(next) = io::next(prev_prefix) {
  next.starts_with(prefix)
} else {
 false
}


but maybe k1 case is not something possible for those storage (same for empty k2 key I am not sure if api allows it). Still needs to check if first consecutive entry is really under prefix.

Api wise I wonder if the api should be more `is_empty_prefix` rather than `contains_prefix`.

Reply:
I am trying to emulate some behavior that could break the function.
```rust
#[crate::storage_alias]
type FooDoubleMapWithEmptyArrays = StorageDoubleMap<Prefix, Identity, [u8; 0], Identity, [u8; 0], [u8; 0]>;```

If the second key is empty and Identity, then the test fails.

```rust
assert_eq!(FooDoubleMapWithEmptyArrays::contains_prefix::<[u8; 0]>([]), false);
FooDoubleMapWithEmptyArrays::insert::<[u8; 0], [u8; 0], [u8; 0]>([], [], []);
assert!(FooDoubleMapWithEmptyArrays::contains_prefix::<[u8; 0]>([]));
FooDoubleMapWithEmptyArrays::remove::<[u8; 0], [u8; 0]>([], []);
assert_eq!(FooDoubleMapWithEmptyArrays::contains_prefix::<[u8; 0]>([]), false);
```